azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * azure-boards-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.azure.boards.common.util;
+
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class AzureFieldsExtractor {
+    private final Gson gson;
+
+    public AzureFieldsExtractor(Gson gson) {
```

Comment:
What is the intended use for this class? It looks like it's only used in the tests.

Reply:
Are we not able to deserialize the response into java objects correctly?

Reply:
It's a utility to extract the actual objects from `fields` fields in certain Azure responses where the `fields` object of the response is dynamic.

Reply:
We cannot deserialize them losslessly.